kuma/static/js/analytics.js

```diff
@@ -1,8 +1,20 @@
 (function(win, doc, $) {
     'use strict';
 
+    // prettier-ignore
     // Adding to globally available mdn object
     var analytics = mdn.analytics = {
+        /**
+         * Handles postMessage events from the interactive editor, passing of
+         * the data to `trackEvent` if the origin of the message is what we expect.
+         * @param {Object} event - The event Object received from the postMessage
+         */
+        interactiveExamplesEvent: function(event) {
+            if (event.origin !== 'https://mdn.github.io') {
+                return false;
+            }
+            this.trackEvent(event.data);
```

Comment:
So, this is the `event` object that is send from `postMessage` and will always only have the following properties:

* data - The data I send back i.e. category, action, label
* origin - Origin of the window that sent the message i.e. the iframe
* source - A reference to the window object of the origin above

One can of course loop over the properties of the `event.data` object and ensure that none of the values are a function. What do you think?